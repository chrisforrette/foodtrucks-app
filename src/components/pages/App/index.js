import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import {
  default as styled,
  ThemeProvider
} from 'styled-components'
import { fetchFoodTrucks } from '../../../actions'
import theme from '../../../styles/theme'
import Header from '../../molecules/Header'
import FoodTruckDetail from '../../molecules/FoodTruckDetail'
import FoodTruckFinder from '../../organisms/FoodTruckFinder'

const AppWrapper = styled.div`
  color: ${({ theme }) => theme.colors.darkGrey};
  display: flex;
  flex-direction: column;
  font-family: ${({ theme }) => theme.fonts.body};
  font-size: 16px;
  height: 100vh;
  line-height: 1.5em;
  width: 100%;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
`

export class AppComponent extends Component {
  componentWillReceiveProps (nextProps) {
    const {
      dispatch,
      foodTrucks,
      mapBoundingBox
    } = nextProps

    if (!foodTrucks && mapBoundingBox) {
      dispatch(fetchFoodTrucks())
    }
  }

  render () {
    const {
      dispatch,
      foodTrucks,
      detailViewOpen,
      detailFoodTruckId
    } = this.props

    return <ThemeProvider theme={theme}>
      <AppWrapper>
        <Header />
        <FoodTruckFinder foodTrucks={foodTrucks} />
        <FoodTruckDetail
          dispatch={dispatch}
          foodTrucks={foodTrucks}
          detailViewOpen={detailViewOpen}
          detailFoodTruckId={detailFoodTruckId} />
      </AppWrapper>
    </ThemeProvider>
  }
}

AppComponent.propTypes = {
  foodTrucks: PropTypes.array,
  dispatch: PropTypes.func
}

export default connect(
  ({ foodTrucks, mapBoundingBox, detailViewOpen, detailFoodTruckId }) => ({ foodTrucks, mapBoundingBox, detailViewOpen, detailFoodTruckId }),
  dispatch => ({ dispatch })
)(AppComponent)