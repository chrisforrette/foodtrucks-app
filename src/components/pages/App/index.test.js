import React from 'react'
import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import App from './'

it('should match snapshot', () => {
  const rendered = mount(<App />)
  expect(toJson(rendered)).toMatchSnapshot()
})
