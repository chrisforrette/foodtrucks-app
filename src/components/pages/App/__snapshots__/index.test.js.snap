// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<ThemeProvider
  theme={
    Object {
      "breakpoints": Object {
        "large": "1200px",
        "medium": "900px",
        "small": "600px",
        "xlarge": "1800px",
      },
      "colors": Object {
        "blue": "#41a6f4",
        "darkGrey": "#333",
        "grey": "#999",
        "lightGrey": "#eee",
      },
      "fonts": Object {
        "body": "\\"HelveticaNeue-Light\\", \\"Helvetica Neue Light\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
        "heading": "\\"HelveticaNeue-Bold\\", \\"Helvetica Neue Bold\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
      },
    }
  }
>
  <styled.div>
    <Connect(Component) />
    <FoodTruckFinder
      foodTrucks={Array []}
    />
    <Component
      detailFoodTruckId={123}
      detailViewOpen={false}
      dispatch={[Function]}
      foodTrucks={Array []}
    />
  </styled.div>
</ThemeProvider>
`;
